### PR TITLE
Silence Clang compiler warnings

### PR DESCRIPTION
**Pull request by insunaa**
## 🍰 Pullrequest

This PR fixes current clang compiler warnings.
Most are caused by switch statements that don't handle every case. Some are evaluations of assignments, some are dangling else statements and one is what Clang sees as an evaluation of an address

### CODE REVIEW

File: src/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -792,7 +792,7 @@ void Spell::AddUnitTarget(Unit* target, uint8 effectMask, CheckException excepti
     targetInfo.heartbeatResistChance = 0;
 
     // Calculate hit result
-    targetInfo.missCondition = (m_ignoreHitResult ? SPELL_MISS_NONE : m_caster->SpellHitResult(target, m_spellInfo, targetInfo.effectMask, m_reflectable, &targetInfo.heartbeatResistChance));
+    targetInfo.missCondition = (m_ignoreHitResult ? SPELL_MISS_NONE : m_caster->SpellHitResult(target, m_spellInfo, targetInfo.effectMask, m_reflectable, &(targetInfo.heartbeatResistChance)));
```
Comment (insunaa) [inline src/game/Spells/Spell.cpp:795]: I'm not sure about this one, can somebody else confirm that this is correct?

File: src/mangosd/CliRunnable.cpp [hunk 2]
```diff
@@ -135,7 +135,7 @@ bool ChatHandler::GetDeletedCharacterInfoList(DeletedInfoList& foundList, std::s
     {
         if (resultChar->GetRowCount() > 100)
         {
-            PSendSysMessage("Too many results %llu. Narrow it down.", resultChar->GetRowCount());
+            PSendSysMessage("Too many results %lu. Narrow it down.", resultChar->GetRowCount());
```
Comment (insunaa) [inline src/mangosd/CliRunnable.cpp:138]: This one is also a bit suspect, as it is dependent on the environment (for example on Linux, `lu` is equivalent to uint64, on MacOS `llu` is equivalent to uint64, so I'm guessing this one should probably remain llu and we'll just ignore that particular warning)

Comment (cyberium) [reply src/mangosd/CliRunnable.cpp:138]: I suggest to cast the result to uint32 and use %u format

File: src/game/Spells/Spell.cpp (hunk 1 above)
Comment (cyberium) [reply src/game/Spells/Spell.cpp:795]: What's the warning message?

Comment (insunaa) [reply src/game/Spells/Spell.cpp:795]:
```
field heartbeatResistChance
Type: uint32
Value = true

// In Spell::TargetInfo
public: uint32 heartbeatResistChance
Address of 'targetInfo.heartbeatResistChance' will always evaluate to 'true'clang(-Wpointer-bool-conversion)
```

File: src/game/AI/ScriptDevAI/scripts/outland/netherstorm.cpp [hunk 3]
```diff
@@ -181,7 +181,7 @@ struct npc_manaforge_spawnAI : public ScriptedAI
         Creature *manaforge;
 
         if (m_manaforgeGuid)
-            if (manaforge = m_creature->GetMap()->GetCreature(m_manaforgeGuid))
+            if ((manaforge = m_creature->GetMap()->GetCreature(m_manaforgeGuid)))
```
Comment (cyberium) [inline src/game/AI/ScriptDevAI/scripts/outland/netherstorm.cpp:184]: I understand that you tried to make compiler happy there.
But this way to test the result of the GetCreature() result is not good to me and its the main issue here.
I would suggest to broke that in 2 instructions

```cpp
if (m_manaforgeGuid)
{
     manaforge = m_creature->GetMap()->GetCreature(m_manaforgeGuid);
     if (manaforge)
     {
          .....
     }
}

```

Comment (insunaa) [reply src/game/AI/ScriptDevAI/scripts/outland/netherstorm.cpp:184]: I had it that way before that, but I saw that several other similar comparisons were doing it in this style, so I adapted it there, too

File: src/game/Spells/Spell.cpp (hunk 1 above)
Comment (cyberium) [reply src/game/Spells/Spell.cpp:795]: This one is an error. SpellHitResult prototype have probably changed over time and this happened. Will summon @killerwife for this. Ignore this for this pr.

Comment (killerwife) [reply src/game/Spells/Spell.cpp]: Fixed on my own accord.

**Comment by cyberium** [comment]
Merged thank you

**Pull request closed without merging**